### ISSUE TITLE
SC1089: bash: coproc "NAME" { ...; } - quoted NAME kills parsing

### ISSUE THREAD

**Issue by iam-TJ** [koalaman/shellcheck#3048]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1083, SC1089
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
coproc COPROC_NAME1 { echo "/directory"  ; }
echo "${COPROC_NAME1[0]}"
coproc "COPROC_NAME2" { echo "/directory"  ; }
cname="COPROC_NAME_$$"
coproc "$cname" { echo "/directory"  ; }
```

#### Here's what shellcheck currently says:
```
Line 4	SC1083: This { is literal. Check expression (missing ;/\n?) or quote it.
Line 4	SC1089: Parsing stopped here. Is this keyword correctly matched up?
```
#### Here's what I wanted or expected to see:

Allow quoted `NAME`

**Comment by koalaman** [comment]
Wow, what an embarrassing deficiency. This was caused by oversimplified code from the early days (Jan 2015) that just hadn't been revisited since. You can now use arbitrary names. Thanks!

**Comment by megaannum** [comment]
See issue: bash coproc #3126
ShellCheck version 0.10.0
$uname -a
Linux olympia 6.11.11-300.fc41.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Dec 5 18:38:25 UTC 2024 x86_64 GNU/Linux
$ bash --version
GNU bash, version 5.2.32(1)-release (x86_64-redhat-linux-gnu)


If I paste the code from SC1089:
#!/usr/bin/env bash 
coproc COPROC_NAME1 { echo "/directory"  ; }
echo "${COPROC_NAME1[0]}"
coproc "COPROC_NAME2" { echo "/directory"  ; }
cname="COPROC_NAME_$$" 
coproc "$cname" { echo "/directory"  ; }

into the file X.sh and run shellcheck I get:
]$ shellcheck X.shIn X.sh line 4:
coproc "COPROC_NAME2" { echo "/directory"  ; }
                      ^-- SC1083 (warning): This { is literal. Check expression (missing ;/\n?) or quote it.                 
                                             ^-- SC1089 (error): Parsing stopped here. Is this keyword correctly matched up?

For more information:
  https://www.shellcheck.net/wiki/SC1083 -- This { is literal. Check expressi...
  https://www.shellcheck.net/wiki/SC1089 -- Parsing stopped here. Is this key...


So, #3048 does not seem to be fixed.

**Comment by wileyhy** [comment]
The fix is in the Master branch at commit 79e43c4 from Sep 7th/8th, 2024,
but the most recent stable release is v0.10.0 from Mar 7, 2024.

> Message ID: ***@***.***>
>
Wiley
